apps/src/code-studio/components/progress/StageLockDialog.jsx

```diff
@@ -0,0 +1,284 @@
+import React from 'react';
+import Radium from 'radium';
+import { connect } from 'react-redux';
+import BaseDialog from '@cdo/apps/templates/BaseDialog';
```

Comment:
Just as a design choice, it seems like you _could_ have some elements in common.  The text of your save/cancel buttons is different, but is there a reason we're not using the `Dialog` button styles?  Part of the motivation behind `Dialog` was to standardize our UI as much as possible.

It does seem like we should have a 'teacher styles' version - the default `Dialog` styles are pretty large and student-friendly, and don't necessarily lend themselves to an information-dense view.  Still, I think things like colors and button position could be more standard - it's odd that you've used call-to-action orange on the assessment steps buttons, but not on the save button, and the gray 'Cancel' button looks like it could be disabled.

It's probable that a more design-y person should be involved in this discussion.  In any case, don't let any of this block the PR.